test/ziggurat/messaging/consumer_test.clj

```diff
@@ -248,51 +237,70 @@
+                                   (when (nil? message-arg)
+                                     (reset! processing-fn-called true)))
+            topic-entity-name    (name topic-entity)]
         (producer/publish-to-dead-queue message)
-        (with-redefs [convert-and-ack-message (fn [_ _ _ _ _] nil)]
+        (with-redefs [consumer/convert-and-ack-message (fn [_ _ _ _ _] nil)]
           (with-open [ch (lch/open connection)]
             (let [queue-name          (get-in (rabbitmq-config) [:dead-letter :queue-name])
                   prefixed-queue-name (str topic-entity-name "_" queue-name)
                   [meta payload]      (lb/get ch prefixed-queue-name false)
-                  _                   (process-message-from-queue ch meta payload topic-entity processing-fn)
+                  _                   (consumer/process-message-from-queue ch meta payload topic-entity processing-fn)
                   consumed-message    (util/get-msg-from-dead-queue-without-ack topic-entity-name)]
               (is (= false @processing-fn-called))
               (is (= consumed-message nil))))))))
   (testing "process-message function should reject and re-queue a message if processing fails. It should also report the error"
     (fix/with-queues {topic-entity {:handler-fn #(constantly nil)}}
-      (let [message       (gen-message-payload topic-entity)
-            processing-fn (fn [message-arg]
-                            (is (= message-arg message))
-                            (throw (Exception. "exception message")))
+      (let [message           (gen-message-payload topic-entity)
+            processing-fn     (fn [message-arg]
+                                (is (= message-arg message))
+                                (throw (Exception. "exception message")))
             topic-entity-name (name topic-entity)
-            report-fn-called?  (atom false)]
+            report-fn-called? (atom false)]
         (with-redefs [report-error (fn [_ _] (reset! report-fn-called? true))]
           (producer/publish-to-dead-queue message)
           (with-open [ch (lch/open connection)]
             (let [queue-name          (get-in (rabbitmq-config) [:dead-letter :queue-name])
                   prefixed-queue-name (str topic-entity-name "_" queue-name)
                   [meta payload]      (lb/get ch prefixed-queue-name false)
-                  _                   (process-message-from-queue ch meta payload topic-entity processing-fn)
+                  _                   (consumer/process-message-from-queue ch meta payload topic-entity processing-fn)
                   consumed-message    (util/get-msg-from-dead-queue-without-ack topic-entity-name)]
               (is (= consumed-message message))
               (is @report-fn-called?))))))))
+
+(deftest convert-and-ack-message-test
+  (testing "While constructing a MessagePayload, adds topic-entity as a keyword and retry-count as 0 if message does not already has :retry-count"
```

Comment:
we don't need this test. the code for this has been removed. Try rebasing the PR with master.

Reply:
What do you mean by it has been removed? It is still in master https://github.com/gojek/ziggurat/blob/master/src/ziggurat/messaging/consumer.clj#L16

Reply:
done